proposals/nnnn-video-streaming-state.md

```diff
@@ -0,0 +1,165 @@
+
+With the new projection apps coming, HMI level itself is not enough to tell when a navigation app or a projection app shall stop video streaming. For example, a driver launches a navigation app first, launches a non-media projection app next. The driver shall still be able to hear turn-by-turn instructions. The navigation app is in HMI level `LIMITED` with `audioStreamingState`=`AUDIBLE`. Because the projection app is streaming, the navigation app shall stop video streaming while it is still in `LIMITED`. (Compare to the case that the driver launches a navigation app and a media app in sequence. The navigation app is in HMI level `LIMITED` with `audioStreamingState`=` ATTENUATED`. It continues streaming video.)
+
+## Proposed solution
+
+We propose to add a new parameter `videoStreamingState` to `onHMIStatus` to explicitly notify the app that it must stop video streaming.
+SDL maintains which app can do video streaming or equivalently we say which app is a video source. If the driver launches a SDL app, SDL knows whether the app uses video service or not depending on the `AppHMIType`. Currently only `NAVIGATION` and `PROJECTION` app use video service. Other types cannot not use video service. Note, an app can have multiple `AppHMIType`s.
+
+Assume there is at most one app that can do video streaming at any time, the following table shows how SDL shall do in different cases.
+
+
+| Case# | Current Video source | SDL App2 comes to FULL | Results | SDL Actions regarding videoStreamingState | 
+| -- | --- | --- | --- | --- |
+| 1 | NONE |  App2 does not use video | no app is the video source | send onHMIStatus to app2 with NOT_STREAMABLE|
+| 2 | NONE |  App2 uses video | App2 is the video source |  send onHMIStatus to app2 with STREAMABLE|
+| 3 | App1 |  App2 does not use video | App1 is **still** the video source | send onHMIStatus to  app2 with NOT_STREAMABLE|
+| 4 | App1 |  App2 uses video | App2 is the **new** video source | send onHMIStatus to app1 with NOT_STREAMABLE & app2 with STREAMABLE|
+| 5 | App1 |  CarPlay/Android Auto is the acitve screen | no SDL app is the video source | send onHMIStatus to app1 with NOT_STREAMABLE |
+
+As before, if the app does not stop video service after receiving `onHMIStatus` with `videoStreamingState`=`NOT_STREAMABLE` for certain time, SDL shall send stop service control frame to the app.
```

Comment:
"SDL shall" should be updated to "SDL Core shall"

Reply:
updated